File: order-service/src/main/java/com/michalmlynarczyk/orderservice/model/entity/Order.java
package com.michalmlynarczyk.orderservice.model.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
import java.util.List;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "orders")
public class Order {

    @Id
    private String id;

    private UUID workshopId;

    private Date createdAt;

    private Date collectedAt;

    private OrderStatus status;

    private Client client;

    private Bike bike;

    private List<Service> services;

    private List<Part> parts;

    public enum OrderStatus {
        NEW,
        IN_PROGRESS,
        DONE,
        COLLECTED
    }
}

File: frontend/bicycle-workshop-management-app/src/app/model/order.ts
export interface Order {
    id: string;
    createdAt: Date;
    collectedAt: Date;
    status: string;
    client: {
        firstName: string;
        lastName: string;
        email: string;
    };
    bike: {
        brand: string;
        model: string;
        color: string;
        frameNumber: string;
        productionYear: number;
        description: string;
    };
}


export interface ExtendedOrder {
    id: string;
    createdAt: Date;
    collectedAt: Date;
    status: string;
    client: {
        firstName: string;
        lastName: string;
        email: string;
    };
    bike: {
        brand: string;
        model: string;
        color: string;
        frameNumber: string;
        productionYear: number;
        description: string;
    };
    services: Service[];
    parts: Part[];
}

export interface Service {
    name: string;
    description: string;
    price: number;
    quantity: number;
}

export interface Part {
    name: string;
    description: string;
    price: number;
    quantity: number;
}